src/cli/com/yetanalytics/persephone/cli/util/args.clj

```diff
@@ -0,0 +1,31 @@
+(ns com.yetanalytics.persephone.cli.util.args
+  (:require [clojure.tools.cli :as cli]))
+
+(defn printerr
+  "Print the `err-messages` vector line-by-line to stderr."
+  [err-messages]
+  (binding [*out* *err*]
+    (run! println err-messages))
+  (flush))
+
+(defn handle-args
+  "Parse `args` based on `cli-options` (which should follow the tools.cli
+   specification) and either return `:error`, print `--help` command and
+   return `:help`, or return the parsed `options` map."
+  [args cli-options]
+  (let [{:keys [options summary errors]}
+        (cli/parse-opts args cli-options)
+        {:keys [help]}
+        options]
+    (cond
+      ;; Display help menu and exit
+      help
+      (do (println summary)
+          :help)
+      ;; Display error message and exit
+      (not-empty errors)
+      (do (printerr errors)
```

Comment:
Also this answers the above question a bit. Oh, side effects.

Reply:
Fun historical note: `printerr` was adapted from the [`bail!`](https://github.com/yetanalytics/datasim/blob/master/src/cli/com/yetanalytics/datasim/main.clj#L119) function in DATASIM CLI but I extracted out the `System/exit` call to `-main` so that it can be called in tests without the whole thing shutting down.

Reply:
The testing angle makes sense